src/approximations/nystrom.jl

```diff
@@ -0,0 +1,104 @@
+# Following the algorithm by William and Seeger, 2001
+# Cs is equivalent to X_mm and C to X_mn
+
+function sampleindex(X::AbstractMatrix, r::Real; obsdim::Integer=defaultobs)
+    0 < r <= 1 || throw(ArgumentError("Sample rate `r` must be in range (0,1]"))
+    n = size(X, obsdim)
+    m = ceil(Int, n*r)
+    S = StatsBase.sample(collect(1:n), m; replace=false, ordered=true)
```

Comment:
If I'm not mistaken (I quickly skimmed through StatsBase), you don't need `collect` here. I think you can just write
```suggestion
    S = StatsBase.sample(1:n, m; replace=false, ordered=true)
```